server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -283,90 +290,99 @@ private Owner refreshOwner(Owner owner) {
         // We need to flush here to make sure our pending persists and merges get pushed to the DB
         // so our reference updates don't fail.
-        this.contentCurator.flush();
+        this.ownerContentCurator.flush();
 
         // Go back through each sub and update content references so we don't end up with dangling,
         // transient or duplicate references on any of the subs' products.
-        for (Subscription sub : subs) {
-            this.updateContentRefs(content, owner, sub.getProduct());
-            this.updateContentRefs(content, owner, sub.getDerivedProduct());
+        // for (Subscription sub : subs) {
+        //     this.updateContentRefs(content, owner, sub.getProduct());
+        //     this.updateContentRefs(content, owner, sub.getDerivedProduct());
 
-            for (Product product : sub.getProvidedProducts()) {
-                this.updateContentRefs(content, owner, product);
-            }
+        //     for (Product product : sub.getProvidedProducts()) {
+        //         this.updateContentRefs(content, owner, product);
+        //     }
 
-            for (Product product : sub.getDerivedProvidedProducts()) {
-                this.updateContentRefs(content, owner, product);
-            }
-        }
+        //     for (Product product : sub.getDerivedProvidedProducts()) {
+        //         this.updateContentRefs(content, owner, product);
+        //     }
+        // }
 
         return changed;
     }
 
-    private void addProductContentToMap(Map<String, Content> contentMap, Product product) {
+    private void addProductContentToMap(Map<String, ContentData> contentMap, ProductData product) {
         if (product == null) {
             return;
         }
 
-        Map<String, Content> mapped = new HashMap<String, Content>();
-        List<ProductContent> duplicates = new LinkedList<ProductContent>();
+        Map<String, ContentData> mapped = new HashMap<String, ContentData>();
+        List<ProductContentData> duplicates = new LinkedList<ProductContentData>();
 
-        for (ProductContent pc : product.getProductContent()) {
-            Content content = pc.getContent();
+        if (product.getProductContent() != null) {
+            for (ProductContentData pcd : product.getProductContent()) {
+                ContentData content = pcd.getContent();
 
-            // Check that this product isn't linking to the same content multiple times...
-            if (mapped.containsKey(content.getId())) {
-                log.warn(
-                    "Multiple references to the same content found on a single product; " +
-                    "discarding duplicate: {} => {}, {}",
-                    product, content, mapped.get(content.getId())
-                );
+                // Check that this product isn't linking to the same content multiple times...
+                // TODO: This probably isn't necessary anymore, since our product DTO is more strict
+                // than our entities
```

Comment:
Ahh, yeah. I've removed this comment, since we're still getting the data from the adapter which may or may not be providing us with subclass that isn't as strict as our base DTO.